internal/sdkv2provider/resource_cloudflare_device_posture_rule_test.go

```diff
@@ -289,6 +289,41 @@ func TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks(t *testing.T)
 	})
 }
 
+func TestAccCloudflareDevicePostureRule_Intune(t *testing.T) {
```

Comment:
i'm getting an acceptance test failure here

```
=== RUN   TestAccCloudflareDevicePostureRule_Intune
    resource_cloudflare_device_posture_rule_test.go:303: Step 1/1 error: Error running apply: exit status 1
        
        Error: error creating Device Posture Rule for account "f037e56e89293a057740de681ac9abbe": Error: Invalid connection id (6117)
        
          with cloudflare_device_posture_rule.rbisfrrpde,
          on terraform_plugin_test.tf line 12, in resource "cloudflare_device_posture_rule" "rbisfrrpde":
          12: resource "cloudflare_device_posture_rule" "rbisfrrpde" {
        
--- FAIL: TestAccCloudflareDevicePostureRule_Intune (0.83s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	41.643s
FAIL
make: *** [testacc] Error 1
```

do we need other setup steps here? or is this something that requires manual setup and we shouldn't run all the time for the default account?

Reply:
The issue was my lack of understanding of the correct usage of connection_id. After making corrections, this test is now passing on my end:
```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareDevicePostureRule_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareDevicePostureRule_SerialNumber
--- PASS: TestAccCloudflareDevicePostureRule_SerialNumber (3.56s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersion
--- PASS: TestAccCloudflareDevicePostureRule_OsVersion (2.94s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersionExtra
--- PASS: TestAccCloudflareDevicePostureRule_OsVersionExtra (3.61s)
=== RUN   TestAccCloudflareDevicePostureRule_LinuxOsDistro
--- PASS: TestAccCloudflareDevicePostureRule_LinuxOsDistro (3.79s)
=== RUN   TestAccCloudflareDevicePostureRule_DomainJoined
--- PASS: TestAccCloudflareDevicePostureRule_DomainJoined (4.12s)
=== RUN   TestAccCloudflareDevicePostureRule_Firewall
--- PASS: TestAccCloudflareDevicePostureRule_Firewall (3.62s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption_RequireAll
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption_RequireAll (4.19s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks (3.82s)
=== RUN   TestAccCloudflareDevicePostureRule_Intune
--- PASS: TestAccCloudflareDevicePostureRule_Intune (4.23s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider      34.437s
```